lib/service/mock_api_service.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:math';
+
+import 'package:dartz/dartz.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton()
+class MockApiService {
+
+  List<String> get items => [for (int i = 0; i < 9999999; i++) 'My List Item nr $i'];
```

Comment:
here we use a huge number for generating list items in order to fake the network request, maybe we can reduce it for a smaller number even in the real code we use pagination for the long lists like this to save the use resource like battery and memory for a better experience
  List<String> get items => [for (int i = 0; i < 4 ; i++) 'My List Item nr $i'];